app/js/enhancedDate.js

```diff
@@ -1,5 +1,133 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function (date) {
+// still maintains access to all globals
+  
+  var setDate = {};
+
+//Stores the passed-in date.  
+  setDate = function (date) {
+      var myDate = date;
+    
+//Do some type-checking to make sure you have a valid type.    
+      if(typeof myDate !== null)
+      {
+//Takes milliseconds after epoch or a date object. 
+        
+//Do some type-checking to make sure you have a valid type. 
+        if (typeof myDate == Date() || typeof myDate == "number"){
+          myDate = date;
+        } else {
+          myDate = false;
+        } 
+      }
+      else{
+        
+//If no value is passed, default to now.       
+        setDate = Date(year, month, day, hours, minutes, seconds, milliseconds);
```

Comment:
You can just say setDate = new Date( ) here for "now". All of the values being passed in to the Date constructor here are undefined, so this will just generate an error.